tripy/nvtripy/frontend/tensor.py

```diff
@@ -209,24 +208,37 @@ def eval(self) -> runtime.MemRefValue:
         from nvtripy.backend.mlir.compiler import Compiler
         from nvtripy.backend.mlir.executor import Executor
         from nvtripy.frontend.trace import Trace
+        from nvtripy.frontend.cache import global_cache
+
+        # Collect inputs
+        inputs = Trace._collect_storage_tensors(self.trace_tensor)
+
+        trace = Trace([self.trace_tensor], inputs=inputs)
+        output_devices = [out.device for out in trace.outputs]
 
-        trace = Trace([self])
-        flat_ir = trace.to_flat_ir()
-        mlir = flat_ir.to_mlir()
+        executable = global_cache.get(trace, devices=output_devices) if config.tripy_eager_cache else None
```

Comment:
Maybe we should check `config.tripy_eager_cache` in the cache's `get`/`set`? Otherwise we would have to do it at each call site

Reply:
would there be other call sites?

Reply:
Possibly. I think it'd be cleaner to handle in the cache anyway though.

Reply:
done